node/light.go

```diff
@@ -142,28 +143,46 @@ func (ln *LightNode) falseValidator() p2p.GossipValidator {
 	}
 }
 
-func (ln *LightNode) newFraudProofValidator() p2p.GossipValidator {
-	return func(fraudProofMsg *p2p.GossipMessage) bool {
-		ln.Logger.Info("fraud proof received", "from", fraudProofMsg.From, "bytes", len(fraudProofMsg.Data))
-		fraudProof := abci.FraudProof{}
-		err := fraudProof.Unmarshal(fraudProofMsg.Data)
+func (ln *LightNode) ProcessFraudProof() {
+	// subscribe to state fraud proof
+	sub, err := ln.fraudService.Subscribe(types.StateFraudProofType)
+	if err != nil {
+		ln.Logger.Error("failed to subscribe to fraud proof gossip", "error", err)
+		return
+	}
+
+	// continuously process the fraud proofs received via subscription
+	for {
```

Comment:
same comment here as the other for loop, good to be explicit about how the loop is being controlled.